components/Modal.tsx

```diff
@@ -0,0 +1,12 @@
+import { HTMLAttributes } from 'react';
+import { Container } from './Container';
+
+export function Modal({ children }: HTMLAttributes<HTMLDivElement>) {
+  return (
+    <Container className="absolute top-[50%] left-[50%] translate-x-[-50%] translate-y-[-50%] ">
+      <Container className="grid w-96 sm:w-96 border border-yellow-300 bg-gray-200">
+        {children}
+      </Container>
+    </Container>
+  );
+}
```

Comment:
Ehdotan Modal-komponentiksi seuraavaa, jossa seuraavia muutoksia:
* div-elementin propsit tyyppi poistettu, kun niitä ei käytetty mihinkään (jäljelle pelkkä children-propsi)
* Container korvattu divillä (Container on identtinen sen kanssa, joten tarpeeton
* Modaalin kuuluu mennä kaiken muun päälle, joten lisätty harmaa huntu muun sisällön päälle ja renderöity `createPortal`in avulla, jotta menee sen hetkisen sisällön ulkopuolelle
* z-indeksien arvoilla 98 ja 99 varmistettu, että menee kaiken muun sisällön päälle.

```suggestion
import { ReactNode } from 'react';
import { createPortal } from 'react-dom';

export function Modal({ children }: { children: ReactNode }) {
  return createPortal(
    <>
      <div className="fixed left-0 top-0 z-[98] h-full w-full bg-black opacity-20" />
      <div className="absolute z-[99] top-[50%] left-[50%] translate-x-[-50%] translate-y-[-50%] ">
        <div className="grid w-96 sm:w-96 border border-yellow-300 bg-white">
          {children}
        </div>
      </div>
    </>,
    document.body,
  );
}
```